src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -42,6 +42,7 @@ internal abstract class ExpressionVariableFinder<TFieldOrLocalSymbol> : CSharpSy
                 case SyntaxKind.IfStatement:
                 case SyntaxKind.SwitchStatement:
                 case SyntaxKind.VariableDeclarator:
+                case SyntaxKind.ConstructorDeclaration:
```

Comment:
Should we also handle `DestructorDeclaration` here and add a `Visit` method as for constructors?